JavaTraining/BillsBurgers/src/Burger.java

```diff
@@ -1,111 +1,54 @@
 public class Burger {
 
-    protected String name;
-    protected double burgerPrice;
-    protected String breadType;
-    protected boolean meat;
-    protected boolean lettuce;
-    protected double lettucePrice = 0.60;
-    protected boolean cheese;
-    protected double cheesePrice = 1;
-    protected boolean onions;
-    protected double onionsPrice = 0.60;
-    protected boolean tomato;
-    protected double tomatoPrice = 0.75;
+    protected BurgerComponent[] baseComponents;
+    protected BurgerComponent[] additionalComponents;
 
-    public Burger(String name, String breadType, double burgerPrice) {
-        this.name = name;
-        this.breadType = breadType;
-        this.burgerPrice = burgerPrice;
-        this.meat = true;
-    }
-
-    public Burger(String name, int burgerPrice) {
-        this.name = name;
-        this.burgerPrice = burgerPrice;
-        this.meat = true;
-
-    }
-    public void setLettuce(boolean lettuce) {
-        this.lettuce = lettuce;
-    }
-
-    public void setCheese(boolean cheese) {
-        this.cheese = cheese;
-    }
+    protected double basePrice;
 
-    public void setOnions(boolean onions) {
-        this.onions = onions;
+    public Burger() {
+        this.basePrice = 5;
```

Comment:
if only the additional components is the difference between this constructor and the next one, could we reduce code duplication somehow?